src/gwt/src/org/rstudio/core/client/command/ShortcutManager.java

```diff
@@ -281,11 +275,8 @@ public void onPreviewNativeEvent(NativePreviewEvent event)
          if (handleKeyDown(event.getNativeEvent()))
          {
             event.cancel();
-            resetKeyBuffer();
-         }
-         else
-         {
-            updateKeyBuffer(event.getNativeEvent());
+            keyBuffer_.clear();
+            events_.fireEvent(new RStudioCommandExecutedEvent());
```

Comment:
This event should probably be renamed, since currently it's only fired in response to command execution via a keyboard shortcut. Alternatively, if possible, it should be inserted in a place so that it really is executed after an AppCommand is executed.